Content.Server/Shuttles/Systems/ArrivalsSystem.cs

```diff
@@ -241,6 +257,41 @@ private void OnPlayerSpawn(PlayerSpawningEvent ev)
         }
     }
 
+    private bool TeleportToMapSpawn(EntityUid player, EntityUid stationId, TransformComponent? transform = null)
+    {
+        if (!Resolve(player, ref transform))
+            return false;
+
+        var points = EntityQuery<SpawnPointComponent, TransformComponent>().ToList();
+        _random.Shuffle(points);
+
+        // Find a spawnpoint on the same map as the player is already on now.
+        foreach (var (spawnPoint, xform) in points)
+        {
+            if (spawnPoint.SpawnType != SpawnPointType.LateJoin || _station.GetOwningStation(spawnPoint.Owner, xform) != stationId)
+                continue;
+
+            // Move the player to the spawnpoint.
+            transform.Coordinates = xform.Coordinates;
+
+            return true;
+        }
+
+        // Spawn them at a passenger location
+        foreach (var (spawnPoint, xform) in points)
+        {
+            if (spawnPoint?.Job?.ID != "Passenger" || _station.GetOwningStation(spawnPoint.Owner, xform) != stationId)
+                continue;
+
+            // Move the player to the spawnpoint.
+            transform.Coordinates = xform.Coordinates;
+
+            return true;
+        }
```

Comment:
All stations have latejoin spawn points already so there isn't really a need to have a hardcoded backup to passengers.

Reply:
Is there any harm in doing so though? I'm coding defensively here I guess.

Reply:
it's unnecessary but moreover, hardcoding in a prototype ID is generally frowned upon.